test/e2e_node/eviction_test.go

```diff
@@ -591,10 +633,32 @@ func runEvictionTest(f *framework.Framework, pressureTimeout time.Duration, expe
 			time.Sleep(30 * time.Second)
 			ginkgo.By("setting up pods to be used by tests")
 			pods := []*v1.Pod{}
+			podsToSucceed := []*v1.Pod{}
 			for _, spec := range testSpecs {
+				if spec.expectToSucceed {
+					podsToSucceed = append(podsToSucceed, spec.pod)
+					continue
+				}
 				pods = append(pods, spec.pod)
 			}
-			e2epod.NewPodClient(f).CreateBatch(ctx, pods)
+
+			e2ePodClient := e2epod.NewPodClient(f)
+
+			// Create all expected pods and wait for them to succeed.
+			// CreateBatch was not used to prevent a race condition because of
+			// underlying call to CreateSync, which may fail if the wait function
+			// misses to see the pod readiness before it terminates.
+			for _, pod := range podsToSucceed {
+				*pod = *e2ePodClient.Create(ctx, pod)
+			}
+
+			ginkgo.By(fmt.Sprintf("Waiting for the expected pods to complete: %v", podsToSucceed))
+			for _, pod := range podsToSucceed {
+				framework.ExpectNoError(e2epod.WaitForPodSuccessInNamespace(ctx, f.ClientSet, pod.Name, pod.Namespace))
+			}
```

Comment:
(optional) It might be cleaner to just launch a bunch of goroutines:

```suggestion
			wg := sync.WaitGroup{}
			wg.Add(len(podsToSucceed))
			for _, pod := range podsToSucceed {
				go func(pod *v1.Pod) {
					defer wg.Done()
					defer ginkgo.GinkgoRecover()
					pod = e2ePodClient.Create(ctx, pod)
					framework.ExpectNoError(e2epod.WaitForPodSuccessInNamespace(ctx, f.ClientSet, pod.Name, pod.Namespace))
				}(pod)
			}
			wg.Wait()
```

It also means that each pod only waits for the maximum timeout. Otherwise, the last pod in the list gets to wait for the maximum timeout plus the aggregate wait time of all pods before it.